src/main/java/me/magnet/consultant/ServiceIdentifier.java

```diff
@@ -47,6 +56,10 @@ public String getServiceName() {
 		return instance;
 	}
 
+	public Set<String> getTags() {
```

Comment:
Return an `ImmutableSet` instead. This allows others to manipulate the contents of the `Set` post-construction.